docs/examples/aws_tags.rb

```diff
@@ -0,0 +1,23 @@
+require 'chef/provisioning/aws_driver'
+
+with_driver 'aws::us-west-2'
+
+machine 'ref-machine-1' do
+  action :allocate
+end
+
+aws_instance 'ref-machine-1' do
```

Comment:
Hmm - I totally didn't think about the fact that most people are going to use `machine` and that doesn't support tag attributes.  This solution seems less than intuitive for people.  And I don't want to add a `tag` attribute to `machine` because that would require a new release of chef-provisioning...  And whatever solution we use will have to work for machine, machine_batch and load_balancer.

Its not awesome, but I think we should read a `tags` field in the machine_options like so:

```ruby
machine 'test' do
  machine_options bootstrap_options: {
    aws_tags: {'key' => 'value'}
  }
end
machine_batch 'test-batch' do
  machine_options bootstrap_options: {
    aws_tags: {'key' => 'value'}
  }
  machine 'test1'
  machine 'test2'
end

load_balancer 'test_lb' do
  load_balancer_options aws_tags: {'key' => 'value'}
end
```

@jkeiser what do you think of this?  Its a weird dual-user-interface problem, but I don't want to add a base `provider_tags` attribute to these resources in chef-provisioning because that implies all providers support tags.

Long term, I want to fix this by having `machine` in a recipe lookup the correct `AwsInstance` resource using the new resource mapping introduced in Chef 12.  Then
```ruby
machine 'ref-machine-1` do
  aws_tags :marco => 'polo'
end
```
would correctly instantiate a `AwsInstance` resource and apply the `aws_tags` to it.